### DISCUSSION TITLE
Is the externals type of module implemented and, if so, why is the import value an object?

### DISCUSSION THREAD

**Discussion by Aaronius** [webpack/webpack#17083] [Q&A]
Is the externals type of `module` supported yet? I saw on the [v5 release notes](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#externals) that it wasn't implemented at that time. However, I see an externals type of `module` [in the docs](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#externals) with no note of it _not_ being supported at this point.

If it isn't supported, is there any way to have an import map like this in my html file:

```
<script type="importmap">
  {
    "imports": {
      "woof": "http://localhost:3002/woof.mjs"
    }
  }
</script>
```

And then import the `woof` module from my application code like this:

```js
import woof from "woof";

console.log(woof);
```

Currently, if I do this, I'm expecting `woof` to be a simple object that looks like `{foo: "bar"}` but instead I'm seeing:

![image](https://user-images.githubusercontent.com/210820/235012522-c1551097-e201-4f22-8edc-a05da911f7cb.png)

If the externals type of `module` isn't supported, I would have expected an error.

Any ideas? Here's my `woof` module, in case it helps:

```js
/******/ // The require scope
/******/ var __webpack_require__ = {};
/******/ 
/************************************************************************/
/******/ /* webpack/runtime/define property getters */
/******/ (() => {
/******/ 	// define getter functions for harmony exports
/******/ 	__webpack_require__.d = (exports, definition) => {
/******/ 		for(var key in definition) {
/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 			}
/******/ 		}
/******/ 	};
/******/ })();
/******/ 
/******/ /* webpack/runtime/hasOwnProperty shorthand */
/******/ (() => {
/******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ })();
/******/ 
/************************************************************************/
var __webpack_exports__ = {};
/* harmony export */ __webpack_require__.d(__webpack_exports__, {
/* harmony export */   "Z": () => (__WEBPACK_DEFAULT_EXPORT__)
/* harmony export */ });
// const woof = () => {
//   console.log("woof");
// };

/* harmony default export */ const __WEBPACK_DEFAULT_EXPORT__ = ({
  foo: "bar"
});
var __webpack_exports__default = __webpack_exports__.Z;
export { __webpack_exports__default as default };
```

**Comment by alexander-akait** [comment]
> Is the externals type of module supported yet? I saw on the [v5 release notes](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#externals) that it wasn't implemented at that time. However, I see an externals type of module [in the docs](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#externals) with no note of it not being supported at this point.

Because it supported.

please create a small reproducible example, because I don't fully undestand what do you want

**Comment by Aaronius** [comment]
Sure. Clone this repo: https://github.com/Aaronius/federated-modules-spike. Run `npm install`. Run `npm start`.

app2 serves an [ES module named woof](https://github.com/Aaronius/federated-modules-spike/blob/import-maps/app2/src/woof.js) at http://localhost:3002/woof.mjs.

app1's [index.html has an import map](http://localhost:3002/woof.mjs) that maps the `woof` bare module name to `http://localhost:3002/woof.mjs`.

From app1, I want to [import the woof module at runtime and log its value to the console](https://github.com/Aaronius/federated-modules-spike/blob/import-maps/app1/src/LandingPage.js#L3-L5). I want to use the `import woof from 'woof';` syntax rather than the `const woof = await import('woof');` syntax.

You can ignore everything else in the apps. Thanks for helping out.

**Reply by alexander-akait** [reply]
Do you want to load external module from one app in other using module federation?

**Reply by Aaronius** [reply]
No, even though the repo has some federated modules-related code, what I'm attempting to do at the moment doesn't have anything to with federated modules. I think I have a better understanding of the right questions to ask now though and I have a more isolated repository to demonstrate what I'm describing. I'm going to close this discussion and open a new one.

**Comment by Aaronius** [comment]
I'm going to close
